src/rpc/client.cpp

```diff
@@ -230,25 +241,35 @@ RPCConvertValues(const std::string &strMethod, const std::vector<std::string> &s
                 WrongNumberOfParams(requiredParams.size(), optionalParams.size(), strParams.size()));
     }
 
-    std::vector<bool> allParams(requiredParams.begin(), requiredParams.end());
+    std::vector<Conversion> allParams(requiredParams.begin(), requiredParams.end());
     allParams.reserve(requiredParams.size() + optionalParams.size());
     allParams.insert(allParams.end(), optionalParams.begin(), optionalParams.end());
 
     for (std::vector<std::string>::size_type idx = 0; idx < strParams.size(); idx++) {
-        const bool shouldConvert = allParams[idx];
+        const Conversion conversion = allParams[idx];
         const std::string& strVal = strParams[idx];
+        auto parsedParam = ParseNonRFCJSONValue(strVal);
```

Comment:
`parsedParam` is not required if `conversion == Conversion::None`. Avoiding calling `ParseNonRFCJSONValue` in that case wouldn't matter except as a micro-optimization, except that `UniValue::read` (together with the functions it uses, `getJsonToken` and `jsonTokenIsValue`) is a non-trivial amount of code and although I don't think it can throw an exception or cause side effects, it's not obvious. The most straightforward change to avoid calling it for `Conversion::None` would be just to duplicate `auto parsedParam = ParseNonRFCJSONValue(strVal);` in the `case` arms for the other two cases.